### PR TITLE
Add option to export/import tasks

### PR DESCRIPTION
**Pull request by Naveen3Singh**
This PR addresses https://github.com/SimpleMobileTools/Simple-Calendar/issues/1666

### Changes
 - Add support for exporting/importing tasks
 - Use UUID as import ID for tasks as well
 - Make exporting events optional
 - Fixed export issue related to future repeating events (repeating events were not exported if `exportPastEvents` was false)

**Note:** Recurrences of a repeating task that were marked completed are not yet exported i.e. when the exported ICS is imported again, those specific recurrences will be marked incomplete.

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/helpers/IcsExporter.kt [hunk 1]
```diff
@@ -153,4 +127,67 @@ class IcsExporter {
...
+            fillDescription(event.description.replace("\n", "\\n"), writer)
+            fillReminders(event, writer, reminderLabel)
+            fillIgnoredOccurrences(event, writer)
+
+            eventsExported++
+            writeLn(END_EVENT)
+        }
+    }
+
+    private fun writeTodo(writer: BufferedWriter, task: Event) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/helpers/IcsExporter.kt:163]: rename it to Task to make it clear

File: app/src/main/res/layout/dialog_export_events.xml [hunk 2]
```diff
@@ -48,15 +48,66 @@
...
+                android:background="@null"
+                android:checked="true"
+                android:clickable="false"
+                android:layoutDirection="rtl"
+                android:padding="@dimen/medium_margin"
+                android:text="@string/export_tasks" />
+
+        </RelativeLayout>
+
+        <ImageView
```
Comment (tibbi) [inline app/src/main/res/layout/dialog_export_events.xml:96]: add an id to all views

**Comment by tibbi** [comment]
cmon, show some error message if both events and tasks are unchecked, at least No items found. Also reduce the checkbox paddings a bit, they are too high now. Make it the same as in the app settings.

**Comment by tibbi** [comment]
events are also missing some UID so at export/import they will be duplicated

**Comment by tibbi** [comment]
actually we have no_entries_for_exporting, so use that when events and tasks are both unchecked, so we dont have to add new strings. But show it right at pressing OK, dont allow selecting a file etc.

**Comment by Naveen3Singh** [comment]
> Also reduce the checkbox paddings a bit, they are too high now. Make it the same as in the app settings.

I thought so too but the paddings are the same as app settings, it just looks taller because the width is lesser here. 
I'm reducing the vertical paddings to 4dp+4dp (from 8dp+8dp)

**Comment by Naveen3Singh** [comment]
> tasks are also missing some UID so at export/import they will be duplicated

right, for some reason, I thought duplicating the events was the default behavior :smile:
 
I'll add a method to migrate existing tasks to use import Ids (for existing users who created tasks)

**Comment by Naveen3Singh** [comment]
Check now please.

**Comment by tibbi** [comment]
those paddings are too small now, just check how are the settings done...

**Comment by Naveen3Singh** [comment]
image

it was the same as the settings. 8dp on top and 8dp on bottom.
I [just](https://github.com/SimpleMobileTools/Simple-Calendar/pull/1937/commits/0725eabcb229c835ccaee2eea65729c0d4f2509e) increased it to 6dp from 4dp.

**Comment by tibbi** [comment]
the settings use 8 dp

**Comment by Naveen3Singh** [comment]
Yes.
I'm saying the dialog also used 8dp when I created the PR, I decreased it to 6dp because you said it looked a bit too big in this situation.

**Comment by Naveen3Singh** [comment]
Yes.
I'm saying the dialog also used 8dp when I created the PR, I decreased it to 6dp because you said it looked a bit too big in this situation.

**Comment by Naveen3Singh** [comment]
Do you want it to be same as the settings even if it looks bigger here?

**Comment by tibbi** [comment]
no need, it is ok like this. Looks perfectly fine for the initial implementation, we will see what feedback will it get. Thanks

**Pull request merged**